### DISCUSSION TITLE
Sitewise: Custom component models failing

### DISCUSSION THREAD

**Discussion by trevransom** [aws/aws-cdk#29463] [General]
### Describe the bug

When trying to implement a custom component model in the AWS CDK for Sitewise in Python it gives the error "Custom composite model must have either id, externalId, or path"

### Expected Behavior

That it would work directly

### Current Behavior

Gives the error "Custom composite model must have either id, externalId, or path"

### Reproduction Steps

```py
class equipmentUnitModel(Construct):
    def __init__(self, scope: Construct, id: str, model_id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)
        [self.my](http://self.my/)_sitewise_model = sitewise.CfnAssetModel(
            self,
            "equipment_unit_model",
            asset_model_name="equipment_unit_model",
            asset_model_description="equipment unit model",
            asset_model_composite_models=[
                sitewise.CfnAssetModel.AssetModelCompositeModelProperty(
                    name="demo",
                    type="CUSTOM"
                ),
            ]
        )
```

### CDK CLI Version

2.131.0 (build 92b912d)

### Node.js Version

v21.6.2

### OS

Mac

### Language

Python

**Comment by pahud** [comment]
>Gives the error "Custom composite model must have either id, externalId, or path"

This seems to be a cloudformation error? Can you share the `cdk synth` output of the `AWS::IoTSiteWise::AssetModel` resource? Looks like some property is required but missing. This doesn't seem to be a CDK bug we just need to figure out what was missing though.

**Reply by trevransom** [reply]
MyCfnAssetComponentModel:
    Type: AWS::IoTSiteWise::AssetModel
    Properties:
      AssetModelExternalId: ComponentModelExternalId
      AssetModelName: assetComponentModelName
      AssetModelType: COMPONENT_MODEL
    Metadata:
      aws:cdk:path: GoogleSitewiseCdkStack/MyCfnAssetComponentModel
  MyCfnAssetModel:
    Type: AWS::IoTSiteWise::AssetModel
    Properties:
      AssetModelCompositeModels:
        - ComposedAssetModelId:
            Ref: MyCfnAssetComponentModel
          ExternalId: CompositeModelexternalId
          Name: CompositeModel1
          Type: CUSTOM
      AssetModelDescription: gj
      AssetModelName: assetModelName

**Reply by trevransom** [reply]
Also getting this error: "Resource handler returned message: "Feature not supported yet (Service: IoTSiteWise," with this cdk synth setup:

 MyCfnAssetModel:
    Type: AWS::IoTSiteWise::AssetModel
    Properties:
      AssetModelCompositeModels:
        - ExternalId: CompositeModelexternalId
          Name: CompositeModel1
          Type: CUSTOM
      AssetModelDescription: gj
      AssetModelName: assetModelName

**Comment by trevransom** [comment]
@pahud any thoughts?

**Comment by gdenn** [comment]
What is the suggested workaround in the meanwhile?

**Reply by gdenn** [reply]
I found one myself using the `CfnAssetModel.add_override(..)` method:

here a small snippet from my code that demonstrates the workaround:

```python
_result = sitewise.CfnAssetModel(
    scope, 
    id,
    asset_model_name=asset_model.name,
    asset_model_description=asset_model.description,
    asset_model_properties=_properties,
    asset_model_composite_models=_composite_models,
    asset_model_hierarchies=_hierarchies,
    tags=_tags
  )

  if asset_model.composite_models is not None:
    for _composite_model in asset_model.composite_models:
      _result.add_override(
        f"Properties.AssetModelCompositeModels.{asset_model.composite_models.index(_composite_model)}.ExternalId",
        _composite_model.logical_id
      )

  return _result
```

**Reply by gdenn** [reply]
Is there a possibility to contribute a PR that adds the Id, ExternalId and Path variables to the `CfnAssetModel.AssetModelCompositeModelProperty`?

I'd be interested to do that.
